pseudocode-defns/pseudocode_defns.tex

```diff
@@ -0,0 +1,490 @@
+    
+Note that, because both \texttt{L} and \texttt{U} are of type \texttt{T}, there is no need to explicitly pass \texttt{T}; the type \texttt{T} can be inferred. \texttt{IntervalDomain} is defined on any type that implements the trait \texttt{TotalOrd}.\footnote{As of June 28, the OpenDP library requires the weaker condition of partial ordering (implements \texttt{PartialOrd}) instead.}
+\end{definition}
+
+For example, \texttt{IntervalDomain(1:u32, 17:u32)} corresponds to a domain that contains all the \texttt{u32} values \texttt{v} such that \texttt{1 <= v} and \texttt{v <= 17}.
+
+\begin{definition}[\texttt{SizedDomain}] $\sfi$
+    \texttt{SizedDomain(D)} is a domain that specifies the length of the enclosed domain \texttt{D}. The length of elements of the enclosed domain is specified with an element of type \texttt{usize}.
+\end{definition}
+
+For example, \texttt{SizedDomain(VectorDomain(AllDomain(u32), n)} corresponds to the domain of all vectors of length $\texttt{n}$ and elements of type \texttt{u32}.
+
+\begin{definition}[\texttt{VectorDomain}]
+\texttt{VectorDomain(D)} is the domain of all vectors of elements drawn from domain \texttt{D}.
+\end{definition}
+
+For example, \texttt{VectorDomain(AllDomain(u32))::Carrier} is \texttt{Vec(u32)}.
+
+\horizline
+
+\subsection{Notes, todos, questions}
+\todonei{Add clampable domain (\texttt{ClampableDomain}) -- waiting until \texttt{TotalOrd} is fully implemented in the OpenDP library.}
+
+\section{Traits}
+% any traits should be defined here
+
+\begin{definition}[\texttt{Bounded}]
+    A type $\T$ has trait \texttt{Bounded} $\iffText$ $\T$ has some upper bound and some lower bound (some smallest possible value and some largest possible value).
+\end{definition}
+
+\begin{definition}[\texttt{DistanceConstant}]
+    A type \texttt{TO} has trait \texttt{DistanceConstant(TI)} $\iffText$
+    \begin{itemize}
+        \item \texttt{TO} has trait \texttt{Mul(Output=TO)} (multiplication can be done with type \texttt{TO})
+        \item \texttt{TO} has trait \texttt{Div(Output=TO)} (some form of inverse mapping can be done with type \texttt{TO})
+        \item \texttt{TO} has trait \texttt{PartialOrd} (\texttt{TO} has a partial ordering)
+        \item \texttt{TO} has trait \texttt{InfCast(TI)}
+    \end{itemize}
+    \inOpenDPRust{DistanceConstant}{https://github.com/opendp/opendp/blob/main/rust/opendp/src/traits.rs}
+\end{definition}
+
+\begin{definition}[\texttt{Domain}]
+A type \texttt{T} has trait \texttt{Domain} $\iffText$ it can represent a set of values that make up a domain. The \texttt{Domain} implementation prescribes a type for members of the domain, as well as a method to check if any instance of that type is a member of that domain.
+\end{definition}
```

Comment:
Seems like this should be in the Domains section.  This is the first indication I'm seeing that Domains are types.

Reply:
The reason we included `Domain` here is because it is a trait that some types have (`Domain` itself is not a domain).

In reference to your second comment, we will be sure to better emphasize the types that various domains have (see [this reply](https://github.com/opendp/whitepapers/pull/4#discussion_r671491645) above).

Reply:
Got it, thanks. Then the beginning (intro text) of the Domains section should say that domains are represented as types with the trait Domain (and cross-reference to this definition).

Reply:
Done